src/tools/decomp.py

```diff
@@ -53,47 +99,284 @@
 
 import math
 import time
+from loadbal import qbicity, changeIndex, halfDecomp, addHsymmetry, adaptNeiList,reDistCellsHom,nodeGridSizeCheck
 
-# need to factorize n to find optimum nodeGrid
-def nodeGrid(n):
-	ijkmax = 3*n*n + 1
-	d1 = 1
-	d2 = 1
-	d3 = 1
-	for i in xrange(1,n+1):
-		for j in xrange(i,n+1):
-			for k in xrange(j,n+1):
-				if (i*j*k == n) and (i*i + j*j + k*k < ijkmax):
-					d1 = k
-					d2 = j
-					d3 = i
-					ijkmax = i*i + j*j + k*k
-	return Int3D(d1,d2,d3)
+__author__ = 'Horacio V Guzman'
+__email__ = 'horacio.v.g at gmail dot com'
+__version__ = '1.0'
+__all__ = [
+    'nodeGrid', 'cellGrid', 'cherrypickTotalProcs',
+    'neiListHom', 'neiListAdress'
+          ]
 
-def cellGrid(box_size, node_grid, rc, skin):
+
+# WARNING! New arguments are needed! At least...box_size,rc,skin
+
+
+
+def nodeGrid(box_size,rc,skin,n,eh_size=0,ratioMS=0,idealGas=0,slabMSDims=[0,0,0]):
```

Comment:
The syntax of ```nodeGrid``` is changed. This will break all existing simulation scripts. Perhaps we should print some message to the user.

Reply:
Is there a way keep it backwards compatible?

Reply:
Yeah, that was the initial idea. However @stuehn, Garching collaborators and myself checked that the new Domain Decomposition algorithm, in particular the processors allocation (nodeGrid) has speedups of around 10% even for homogenous-cubic systems. Due to the Qubicity feature that minimally requires Box_size,rc,skin,n

Reply:
Still I would really prefer having at least a backwards compatible failback! Could we use function polymorphism? If you only hand in `n` it uses the old version and issues a warning.

Reply:
Or leave `nodeGrid()` as is and create a new function `nodeGrid_ HeSpaDDA()`!